src/alert/src/Alert.tsx

```diff
@@ -1,78 +1,69 @@
-import React, { PureComponent } from 'react'
-import PropTypes from 'prop-types'
-import { spacing, dimensions, position, layout } from 'ui-box'
-import { withTheme } from '../../theme'
-import { Pane } from '../../layers'
-import { Heading, Paragraph } from '../../typography'
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import Box, { BoxProps } from 'ui-box'
+
 import { IconButton } from '../../buttons'
+import { IntentType } from '../../constants'
 import { Icon } from '../../icon'
+import { Pane } from '../../layers'
+import { PropsWithTheme, withTheme } from '../../theme'
+import { AnyFunction } from '../../types/helper'
+import { Heading, Paragraph } from '../../typography'
 
-class Alert extends PureComponent {
-  static propTypes = {
-    /**
-     * Composes some Box APIs.
-     */
-    ...spacing.propTypes,
-    ...position.propTypes,
-    ...layout.propTypes,
-    ...dimensions.propTypes,
-
-    /**
-     * The content of the alert. When a string is passed it is wrapped in a `<Text size={400} />` component.
-     */
-    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+type Appearance = 'default' | 'card'
 
-    /**
-     * The intent of the alert.
-     */
-    intent: PropTypes.oneOf(['none', 'success', 'warning', 'danger'])
-      .isRequired,
+interface IProps extends Partial<BoxProps> {
+  // The content of the alert. When a string is passed it is wrapped in a `<Text size={400} />` component.
+  children?: string | React.ReactNode
 
-    /**
-     * The title of the alert.
-     */
-    title: PropTypes.node,
+  // The intent of the alert.
+  intent?: IntentType
 
-    /**
-     * When true, show a border on the left matching the type.
-     */
-    hasTrim: PropTypes.bool,
+  // The title of the alert.
+  title?: React.ReactNode
 
-    /**
-     * When true, show a icon on the left matching the type,
-     */
-    hasIcon: PropTypes.bool,
+  // When true, show a border on the left matching the type.
+  hasTrim?: boolean
 
-    /**
-     * When true, show a remove icon button.
-     */
-    isRemoveable: PropTypes.bool,
+  // When true, show a icon on the left matching the type,
+  hasIcon?: boolean
 
-    /**
-     * Function called when the remove button is clicked.
-     */
-    onRemove: PropTypes.func,
+  // When true, show a remove icon button.
+  isRemoveable?: boolean
+
+  // Function called when the remove button is clicked.
+  onRemove?: AnyFunction
```

Comment:
```suggestion
  onRemove?: () => void
```

Using an alias like `AnyFunction` is a code smell and is annoying to update if you end up passing something later. It is also more explicit.

Reply:
Thanks! I'm updating `any` related typing in my next commit :)